File: app/src/main/java/com/example/whattowatch/model/mappers/CastMapper.java
package com.example.whattowatch.model.mappers;

import com.example.whattowatch.model.mymodel.MyCastModel;
import com.example.whattowatch.model.response.cradits.Cast;

import java.util.ArrayList;
import java.util.List;

public class CastMapper {
    public static MyCastModel convertMyCastModel(Cast cast){
        return new MyCastModel(cast.getId(),
                               cast.getCharacter(),
                               cast.getName(),
                               cast.getProfilePath());
    }
    public static List<MyCastModel> convertListMyCastModel(List<Cast> casts){
        List<MyCastModel> myCastModels = new ArrayList<>();
        for(Cast cast : casts){
            myCastModels.add(convertMyCastModel(cast));
        }
        return myCastModels;
    }
}

File: app/src/main/java/com/example/whattowatch/model/mappers/CastMapper.kt
package com.example.whattowatch.model.mappers

import com.example.whattowatch.model.mymodel.MyCastModel
import com.example.whattowatch.model.response.cradits.Cast
import java.util.*

object CastMapper {
    fun convertMyCastModel(cast: Cast): MyCastModel {
        return MyCastModel(cast.id ?: 0,
                cast.character,
                cast.name,
                cast.profilePath)
    }

    fun convertListMyCastModel(casts: List<Cast>): List<MyCastModel> {
        val myCastModels: MutableList<MyCastModel> = ArrayList()
        for (cast in casts) {
            myCastModels.add(convertMyCastModel(cast))
        }
        return myCastModels
    }
}